functions/models/src/fhir/fhirObservation.ts

```diff
@@ -149,35 +149,52 @@ export const fhirObservationConverter = new Lazy(
           effectivePeriod: optionalish(
             z.lazy(() => fhirPeriodConverter.value.schema),
           ),
-          effectiveDateTime: optionalish(dateConverter.schema),
-          effectiveInstant: optionalish(dateConverter.schema),
+          effectiveDateTime: optionalish(dateConverterISO.schema),
+          effectiveInstant: optionalish(dateConverterISO.schema),
         })
         .transform((values) => new FHIRObservation(values)),
-      encode: (object) => ({
-        ...fhirResourceConverter.value.encode(object),
-        status: object.status,
-        code: fhirCodeableConceptConverter.value.encode(object.code),
-        component:
-          object.component?.map(
-            fhirObservationComponentConverter.value.encode,
-          ) ?? null,
-        valueQuantity:
-          object.valueQuantity ?
-            fhirQuantityConverter.value.encode(object.valueQuantity)
-          : null,
-        effectivePeriod:
-          object.effectivePeriod ?
-            fhirPeriodConverter.value.encode(object.effectivePeriod)
-          : null,
-        effectiveDateTime:
-          object.effectiveDateTime ?
-            dateConverter.encode(object.effectiveDateTime)
-          : null,
-        effectiveInstant:
-          object.effectiveInstant ?
-            dateConverter.encode(object.effectiveInstant)
-          : null,
-      }),
+      encode: (object) => {
+        const base = {
+          ...fhirResourceConverter.value.encode(object),
+          status: object.status,
+          code: fhirCodeableConceptConverter.value.encode(object.code),
+          component:
+            object.component?.map(
+              fhirObservationComponentConverter.value.encode,
+            ) ?? null,
+          valueQuantity:
+            object.valueQuantity ?
+              fhirQuantityConverter.value.encode(object.valueQuantity)
+            : null,
+        }
+
+        if (object.effectivePeriod) {
```

Comment:
thanks for implementing the mutual exclusivity here! there's also an `effectiveTiming` field, but i'm not sure if that actually gets used by MHC...